src/nvim/regexp_nfa.c

```diff
@@ -4844,8 +4846,12 @@ static int nfa_regmatch(nfa_regprog_T *prog, nfa_state_T *start, regsubs_T *subm
   /* Some patterns may take a long time to match, especially when using
    * recursive_regmatch(). Allow interrupting them with CTRL-C. */
   fast_breakcheck();
-  if (got_int)
-    return FALSE;
+  if (got_int) {
+    return FALSE;  // NOLINT(readability/bool)
+  }
+  if (nfa_time_limit != NULL && profile_passed_limit(*nfa_time_limit)) {
+      return FALSE;  // NOLINT(readability/bool)
+  }
```

Comment:
Why do we need to suppress these linter errors? If I'm not mistaken, we can just replace `FALSE` with `false`.